### PR TITLE
[all] Update to k/k v1.21.0

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
This PR updates k/k to v1.21.0. Some of the test packages have been moved different location in latest k/k repo.
This PR is to sync with the same before adding any new tests.

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ramineni** [comment]
@lingxiankong @jichenjc PTAL ^^

**Comment by jichenjc** [comment]
/lgtm

thanks~

**Comment by chrigl** [comment]
/test all

wondering why tests don't kick in 🤔

**Comment by lingxiankong** [comment]
The job cloud-provider-openstack-acceptance-test-lb-octavia failed because it's running against a v1.19 cluster(in Catalyst Cloud), it failed with:

```
I0407 02:56:49.579004       1 controller.go:341] Ensuring load balancer for service octavia-lb-test/test
I0407 02:56:49.751166       1 controller.go:830] Adding finalizer to service octavia-lb-test/test
I0407 02:56:49.892506       1 event.go:291] "Event occurred" object="octavia-lb-test/test" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
W0407 02:56:50.956273       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
W0407 02:56:51.463011       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
W0407 02:56:52.475108       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
W0407 02:56:54.489315       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
E0407 02:56:54.489466       1 event.go:273] Unable to write event: '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"test.167373fa35e22913", GenerateName:"", Namespace:"octavia-lb-test", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Service", Namespace:"octavia-lb-test", Name:"test", UID:"26b597c7-aee2-4270-9f25-7044c04e0252", APIVersion:"v1", ResourceVersion:"29074319", FieldPath:""}, Reason:"EnsuringLoadBalancer", Message:"Ensuring load balancer", Source:v1.EventSource{Component:"service-controller", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xc01365bc6618ff13, ext:29091577586510, loc:(*time.Location)(0x2f659c0)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xc01365bc6618ff13, ext:29091577586510, loc:(*time.Location)(0x2f659c0)}}, Count:1, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'Post https://10.254.0.1:443/api/v1/namespaces/octavia-lb-test/events: failed
to get token for kube-system/service-controller: timed out waiting for the condition'(may retry after sleeping)
```

@ramineni Have you tested OCCM function manually in your cluster? If yes, we could just override the job temporarily, otherwise, I can help test in mine if you have an available container image.

**Comment by lingxiankong** [comment]
for cloud-provider-openstack-e2e-test-csi-cinder, I can see:

```
2021-03-31 11:30:24.280634 | ubuntu-focal-citynetwork | Ran 54 of 166 Specs in 5471.280 seconds
2021-03-31 11:30:24.280644 | ubuntu-focal-citynetwork | FAIL! -- 51 Passed | 3 Failed | 0 Pending | 112 Skipped
```

Have you checked the failed tests?

**Comment by ramineni** [comment]
> for cloud-provider-openstack-e2e-test-csi-cinder, I can see:
> 
> ```
> 2021-03-31 11:30:24.280634 | ubuntu-focal-citynetwork | Ran 54 of 166 Specs in 5471.280 seconds
> 2021-03-31 11:30:24.280644 | ubuntu-focal-citynetwork | FAIL! -- 51 Passed | 3 Failed | 0 Pending | 112 Skipped
> ```
> 
> Have you checked the failed tests?

I'm looking into it.

**Comment by ramineni** [comment]
> The job cloud-provider-openstack-acceptance-test-lb-octavia failed because it's running against a v1.19 cluster(in Catalyst Cloud), it failed with:
> 
> ```
> I0407 02:56:49.579004       1 controller.go:341] Ensuring load balancer for service octavia-lb-test/test
> I0407 02:56:49.751166       1 controller.go:830] Adding finalizer to service octavia-lb-test/test
> I0407 02:56:49.892506       1 event.go:291] "Event occurred" object="octavia-lb-test/test" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
> W0407 02:56:50.956273       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
> W0407 02:56:51.463011       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
> W0407 02:56:52.475108       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
> W0407 02:56:54.489315       1 client_builder_dynamic.go:187] get token failed: the server could not find the requested resource
> E0407 02:56:54.489466       1 event.go:273] Unable to write event: '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"test.167373fa35e22913", GenerateName:"", Namespace:"octavia-lb-test", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Service", Namespace:"octavia-lb-test", Name:"test", UID:"26b597c7-aee2-4270-9f25-7044c04e0252", APIVersion:"v1", ResourceVersion:"29074319", FieldPath:""}, Reason:"EnsuringLoadBalancer", Message:"Ensuring load balancer", Source:v1.EventSource{Component:"service-controller", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xc01365bc6618ff13, ext:29091577586510, loc:(*time.Location)(0x2f659c0)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xc01365bc6618ff13, ext:29091577586510, loc:(*time.Location)(0x2f659c0)}}, Count:1, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'Post https://10.254.0.1:443/api/v1/namespaces/octavia-lb-test/events: failed
> to get token for kube-system/service-controller: timed out waiting for the condition'(may retry after sleeping)
> ```
> 
> @ramineni Have you tested OCCM function manually in your cluster? If yes, we could just override the job temporarily, otherwise, I can help test in mine if you have an available container image.

I didnt test specific functionality , only overall its running or not. Anyway 1.21-rc0 is cut, let me update to it and check.

**Comment by chrigl** [comment]
/test cloud-provider-openstack-e2e-test-csi-cinder

fire off the test again to check the upload to our gs bucket.

File: cmd/openstack-cloud-controller-manager/main.go [hunk 1]
```diff
@@ -23,115 +23,46 @@ import (
...
 
-	fs := command.Flags()
-	namedFlagSets := s.Flags(controllerList, app.ControllersDisabledByDefault.List())
-	for _, f := range namedFlagSets.FlagSets {
-		fs.AddFlagSet(f)
-	}
-
-	fs.BoolVar(&versionFlag, "version", false, "Print version and exit")
+	fss := cliflag.NamedFlagSets{}
+	command := app.NewCloudControllerManagerCommand(ccmOptions, cloudInitializer, app.DefaultInitFuncConstructors, fss, wait.NeverStop)
```
Comment (zetaab) [inline cmd/openstack-cloud-controller-manager/main.go:55]: is this going to change cobra.command from `openstack-cloud-controller-manager` to `cloud-controller-manager`? if yes, we should perhaps modify manifests also?

**Comment by lingxiankong** [comment]
The job `cloud-provider-openstack-acceptance-test-keystone-authentication-authorization` failed because the kube-apiserver complains about communication with the auth webhook which is not installed yet. The job should be refactored in the same way of how the job `cloud-provider-openstack-acceptance-test-lb-octavia` is defined.

At this stage, I will test manually in my v1.21 cluster and override the job if everything looks ok.

**Comment by lingxiankong** [comment]
looking good

/override openlab/cloud-provider-openstack-acceptance-test-keystone-authentication-authorization

**Comment by lingxiankong** [comment]
also tested occm in my v1.21 cluster.

/override openlab/cloud-provider-openstack-acceptance-test-lb-octavia

**Comment by lingxiankong** [comment]
Now I'm looking at the occm version issue.

File: cmd/openstack-cloud-controller-manager/main.go (hunk 1 above)
Comment (lingxiankong) [reply cmd/openstack-cloud-controller-manager/main.go:55]: Nope, the command is not changed, but the help message will look weird as it's saying:

```
Usage:
  cloud-controller-manager [flags]
```

Comment (lingxiankong) [reply cmd/openstack-cloud-controller-manager/main.go:55]: @ramineni I don't think we could resolved the `--version` issue perfectly as it's defined in `NewCloudControllerManagerCommand` method. 

```
$ ./openstack-cloud-controller-manager --version
Kubernetes v0.0.0-master+$Format:%H$
```

One workaround I can think of is to define a different param (e.g. `occm-version`) to maintain our own version, or we leave it to k8s community to provide a mechanism for cloud providers. What do you think?

cc @jichenjc

**Comment by lingxiankong** [comment]
@ramineni we could merge this one and propose PR to fix the version stuff later on.

/lgtm

File: cmd/openstack-cloud-controller-manager/main.go (hunk 1 above)
Comment (ramineni) [reply cmd/openstack-cloud-controller-manager/main.go:55]: > @ramineni I don't think we could resolved the --version issue perfectly as it's defined in NewCloudControllerManagerCommand method.

@lingxiankong right. From here it looked to me that it should take the values from ldflags , https://github.com/kubernetes/component-base/blob/master/version/version.go#L29 , but doesnt seem to work .

> One workaround I can think of is to define a different param (e.g. occm-version) to maintain our own version, or we leave it to k8s community to provide a mechanism for cloud providers. What do you think?

@lingxiankong +1, I checked a lil bit on this before , couldnt get it working as not clean mechanism provided yet . Ill get to this option again . Let me also check with cloudprovider team if they have any other options. Thanks.
I'll raise a bug to track this issue in meantime

Comment (jichenjc) [reply cmd/openstack-cloud-controller-manager/main.go:55]: >>>One workaround I can think of is to define a different param (e.g. occm-version) to maintain our own version, or we leave it to k8s community to provide a mechanism for cloud providers. What do you think?

+1 to me as well, we can provide some quick solution for now and overtime let's switch to cloud-provider way

**Pull request merged**